assets/js/base/components/checkout/form-step/index.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import Label from '../../label';
+
+const FormStep = ( {
+	id,
+	className,
+	stepNumber,
+	title,
+	description,
+	children,
+	StepHeadingContent,
+} ) => {
+	return (
+		<div
+			className={ classnames( className, 'wc-components-checkout-step' ) }
+			id={ id }
+		>
+			<StepNumber stepNumber={ stepNumber } />
+			<StepHeading
+				title={ title }
+				StepHeadingContent={ StepHeadingContent }
+			/>
+			<span className="wc-components-checkout-step__description">
+				{ description }
+			</span>
+			<div className="wc-components-checkout-step__content">
+				{ children }
+			</div>
+		</div>
+	);
+};
+
+const StepNumber = ( { stepNumber } ) => {
```

Comment:
Minor detail, I'd usually place these internal components at the top of the file, above the main component `FormStep` which depends on them. Not a blocker, just exploring code style conventions :)